runtime/autoload/man.vim

```diff
@@ -68,19 +66,63 @@ function! man#read_page(ref) abort
     " call to s:error() is unnecessary
     return
   endtry
-  call s:read_page(path)
+  let page = s:get_page(path)
```

Comment:
Once you move s:get_page into the above try, the if empty(page) also becomes unnecessary. Same with the call in man#open_page.